lib/puppet/settings.rb

```diff
@@ -1500,7 +1495,7 @@ def initialize(name, defaults)
     def_delegator :@values, :[], :lookup
 
     def set(name, value)
-      default = @defaults[name]
+      default = @defaults[name.to_sym]
```

Comment:
Maybe this should not be changed, remove `to_s` form https://github.com/puppetlabs/puppet/pull/8464/files#diff-88049b67b11a0279f42d379d4241b81ac5eab99c111a6c8ac8d41162f364c432R5 and find out where the name needs to be a string